### DISCUSSION TITLE
Proposal - multi-level, hierarchy docs & assets

### DISCUSSION THREAD

**Discussion by Taraskin** [arthelokyo/astrowind#13] [Ideas]
First - thanks for such a great Astro template!
Idea - to have support have multi-level hierarchy for documents & assets (partially done in personal site, will try to PR later).

**Comment by prototypa** [comment]
Hello @Taraskin,
First of all, thank you for contributing to this project.

About the idea, I've been thinking about it for a while but I still don't quite understand what you mean.
Can you give me more details please?

**Reply by Taraskin** [reply]
Hi, @prototypa 
Apologies for being silent - was a bit overloaded with daily routines.
I'll try to explain - I meant support for images hierarchy in **assets/images**.
Original code (for previous version):
`export const findImage = async (imageRoute) => {
  const images = import.meta.glob("../assets/images/*");
  ...
`
I made a change in my personal project to:
`export const findImage = async (imageRoute) => {
  const images = import.meta.glob("../assets/images/**");
  ...
`
Hope, you got the point.
Also, I found it would be nice to have a template/view for documents, other than posts/pages, i.e.:
`  const posts = import.meta.glob("../data/posts/**/*.{md,mdx}", {
`
- handles docs specifically from the `posts` folder, I had an intention to add something outside it, or even - to have 2 separate docs hierarchies - `posts` and `news` (just an example). Not sure if I am going into right direction for docs grouping & managements, just highlighted my thoughts :)

**Reply by prototypa** [reply]
Hi @Taraskin 
Thanks for write. I've also been a bit busy with daily routines, I'm back with updates to the template.

I totally agree with what you say about the images. I'll make the adjustments right now.

As for the other thing you mention (posts and news), there are many ways to do that. Already in the blog I added the features of tags and categories. I'm still not happy with that, there are many ways and I don't want to overcomplicate a generic template either. It takes a little more work, but using combinations of categories and removing some urls it is possible to solve your problem.

I will think about it in depth and then write to you.
